### PR TITLE
Allow moving account data and receipts streams off master

### PR DESCRIPTION
**Pull request by erikjohnston**
We do both of these so that we can move off the `/read_marker` API

Should be reviewable commit by commit

### CODE REVIEW

File: synapse/replication/http/account_data.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+class ReplicationUserAccountDataRestServlet(ReplicationEndpoint):
+    """Add user account data on the appropriate account data worker.
+
+    Request format:
+
+        POST /_synapse/replication/add_user_account_data/:user_id/:type
+
+        {
+            "account_data_type": "m.direct",
```
Comment (clokep) [inline synapse/replication/http/account_data.py:32]: I think `account_data_type` is in the URL, not the content.

This is also true for `ReplicationRoomAccountDataRestServlet`.

File: synapse/config/workers.py [hunk 2]
```diff
@@ -56,6 +56,12 @@ class WriterLocations:
     to_device = attr.ib(
         default=["master"], type=List[str], converter=_instance_to_list_converter,
     )
+    account_data = attr.ib(
+        default=["master"], type=List[str], converter=_instance_to_list_converter,
+    )
+    receipts = attr.ib(
+        default=["master"], type=List[str], converter=_instance_to_list_converter,
+    )
```
[the remark this answers is not in the document]
Comment (erikjohnston) [reply synapse/config/workers.py:64]: Yes, exactly. I think the only things that stop it are a) fix the replication stream store functions to include instance name, and b) change the `StreamToken` to use a vector clock for receipts/account_data.

File: synapse/storage/databases/main/account_data.py [hunk 3]
```diff
@@ -32,27 +34,71 @@
...
+        else:
+            self._can_write_to_account_data = True
+
+            # We shouldn't be running in worker mode with SQLite, but its useful
+            # to support it for unit tests.
+            #
+            # If this process is the writer than we need to use
+            # `StreamIdGenerator`, otherwise we use `SlavedIdTracker` which gets
+            # updated over replication. (Multiple writers are not supported for
+            # SQLite).
```
[the remark this answers is not in the document]
Comment (clokep) [reply synapse/storage/databases/main/account_data.py:72]: (If we can...probably best to do it as a follow-up, but between this, the HTTP endpoints, etc. it seems like a ton of boilerplate.)

**Review by clokep — APPROVED** [review]
I think this looks fine.

**Pull request merged**